src/PublicGoodsDonator.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {FilAddress} from "shim/FilAddress.sol";
+import {OwnedClaimable} from "shim/OwnedClaimable.sol";
+import {IPreStake} from "src/Interfaces/IPreStake.sol";
+import {IWFIL} from "src/Interfaces/IWFIL.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/**
+ * @title PublicGoodsDonator
+ * @author GLIF
+ * @notice The PublicGoodsDonator contract is used as a proxy over the Infinity Pool to donate a % of iFIL tokens to public goods
+ */
+contract PublicGoodsDonator is OwnedClaimable {
+
+  using FilAddress for address;
+
+  event Donate(address indexed account, uint256 donationAmount);
+  event WithdrawFunds(address indexed wallet, uint256 amount);
+
+  /// @dev WAD is used to compute the split % to pg wallet
+  uint256 immutable DENOM = 1e18;
+
+  /// @dev modifier to ensure `donationPercentage` is 100% max
+  modifier noOverDonations(uint256 donationPercentage) {
+    if (donationPercentage > DENOM) {
+      revert InvalidParams();
+    }
+    _;
+  }
+
+  IPreStake public immutable preStake;
+
+  IWFIL private wFIL;
+  IERC20 private iFIL;
+
+  constructor(
+    address _owner,
+    IPreStake _preStake,
+    IWFIL _wFIL,
+    IERC20 _poolToken
+  )
+    OwnedClaimable(_owner)
+  {
+    preStake = _preStake;
+    wFIL = _wFIL;
+    iFIL = _poolToken;
+  }
+
+  /**
+   * @notice Deposit WFIL into the PreStake contract
+   * @param recipient The account to pull WFIL from and forward the iFIL tokens to
+   * @param amount The amount of WFIL stake
+   * @param donationPercent The percentage of the amount to split to the PG wallet. 100 = 100%
+   */
+  function deposit(
```

Comment:
The assumption here is that the ``donationPercent`` is the percent of ``amount`` that this contract will _retain_ and the rest will get set to the recipient, correct?

Reply:
Correct